### ISSUE TITLE
Executing in terminal doesn't work

### ISSUE THREAD

**Issue by es20490446e** [lxqt/pcmanfm-qt#1842]
### Screencast
- https://youtu.be/fkgvekqCFyo

### Settings
- https://pastebin.com/raw/6S2KJa1R

### System
* pcmanfm-qt 1.4.0
* liblxqt 1.4.0
* qt5-tools 5.15.1
* Arch Linux

**Comment by tsujan** [comment]
Not reproducible.

What is `test.sh`?

**Comment by tsujan** [comment]
> It's just:

Then the terminal is shown in a fraction of second and closed. Some window managers might not even show it if it's closed immediately.

"Execute in Terminal" has nothing to do with keeping the terminal open after the execution is finished. If you use this script

```sh
#! /bin/bash
echo start
sleep 5
echo end
exit
```

then the terminal will be opened, you'll see "start" in it, and after 5 seconds it'll be closed, not letting you see "end".

But if you use this script

```sh
#! /bin/bash
echo start
sleep 5
echo end
read
exit
```

the terminal will remain open after 5 seconds and you'll see "end" too.

In short, it's up to the user.

**Comment by tsujan** [comment]
As you already know, you could either **open** a terminal or **execute** in a terminal by using pcmanfm-qt. These are two different functionalities.

Scripts that need to control when the terminal is closed in the second case should do it themselves (`read` is a simple method). Not all terminals have a "hold" option.

As a side note, pcmanfm-qt tries to keep the terminal open if it has a "hold" option *and* the desktop entry of an app has the key `X-KeepTerminal`. However, that key isn't standard, and almost no app uses it. It isn't related to "Execute in Terminal"  either.

**Comment by tsujan** [comment]
pcmanfm-qt can't extend execution beyond exiting.

**Comment by tsujan** [comment]
> The graphic terminal emulator (like qterminal) or the actual terminal (like bash)?

The graphic terminal and when executing a desktop entry with `Terminal=true` and `X-KeepTerminal=true` in it. But, as I said, `X-KeepTerminal` is nonstandard and only works with terminals that have a "hold" option. So, ignore it.

Also see https://github.com/lxqt/qterminal/issues/554#issuecomment-471252822, which is essentially what I said here.
